spotbugs/src/main/java/edu/umd/cs/findbugs/ba/SignatureParser.java

```diff
@@ -33,6 +33,8 @@
  * A simple class to parse method signatures.
  *
  * @author David Hovemeyer
+ *
+ * @deprecated This signature parser does not support generics, use GenericSignatureParser instead.
```

Comment:
Can you also add deprecated annotation to match javadoc?  Or if that only applies to generics, it should us deprecated javadoc annotation.

Reply:
I've added the annotation.

This is a digression but I think we should consider requiring JDK soon (when it's GA), so we can use [JEP-484](https://openjdk.org/jeps/484)'s classfile API: among other things it has a signature parser and it would be a good replacement for these limited parsers.

Reply:
> I've added the annotation.
> 
> This is a digression but I think we should consider requiring JDK soon (when it's GA), so we can use [JEP-484](https://openjdk.org/jeps/484)'s classfile API: among other things it has a signature parser and it would be a good replacement for these limited parsers.

We just updated to Java 11 requirement, let us bump it to Java 17 and then to 21 first. After that we might consider to require Java 25 (next LTS), which is however not released yet... So I don't expect this API be available soon for SpotBugs.